AVOutput/AVOutputTV.cpp

```diff
@@ -0,0 +1,7556 @@
+    {
+        LOGINFO("Entry\n");
+        std::string value;
+        tvDisplayMode_t mode = tvDisplayMode_16x9;
+        std::string pqmode;
+        std::string source;
+        std::string format;
+
+
+        value = parameters.HasLabel("zoomMode") ? parameters["zoomMode"].String() : "";
+        returnIfParamNotFound(parameters,"zoomMode");
+
+        if (validateInputParameter("AspectRatio",value) != 0) 
+	{
+            LOGERR("%s: Range validation failed for AspectRatio\n", __FUNCTION__);
+            returnResponse(false);
+        }
+
+        if (parsingSetInputArgument(parameters, "AspectRatio",source, pqmode, format) != 0)
+        {
+            LOGERR("%s: Failed to parse the input arguments \n", __FUNCTION__);
+            returnResponse(false);
+        }
+
+	if( !isCapablityCheckPassed( pqmode, source, format, "AspectRatio" )) 
+	{
+            LOGERR("%s: CapablityCheck failed for AspectRatio\n", __FUNCTION__);
+            returnResponse(false);
+        }
+
+        if(!value.compare("TV 16X9 STRETCH")) 
+	{
+            mode = tvDisplayMode_16x9;
+        }
+        else if (!value.compare("TV 4X3 PILLARBOX"))
+	{
+            mode = tvDisplayMode_4x3;
+        }
+        else if (!value.compare("TV NORMAL"))
+	{
+            mode = tvDisplayMode_NORMAL;
+        }
+        else if (!value.compare("TV DIRECT"))
+	{
+            mode = tvDisplayMode_DIRECT;
+        }
+        else if (!value.compare("TV AUTO"))
+	{
+            mode = tvDisplayMode_AUTO;
+        }
+        else if (!value.compare("TV ZOOM"))
+	{
+            mode = tvDisplayMode_ZOOM;
+        }
+        else 
+	{
+            returnResponse(false);
+        }
+        m_videoZoomMode = mode;
+        tvError_t ret = setAspectRatioZoomSettings (mode);
+
+        if(ret != tvERROR_NONE) 
+	{
+            returnResponse(false);
+        }
+        else 
+	{
+            //Save DisplayMode to localstore and ssm_data
+            int params[3]={0};
+            params[0]=mode;
+            int retval=updatePQParamsToCache("set","AspectRatio",pqmode,source,format,PQ_PARAM_ASPECT_RATIO,params);;
+
+            if(retval != 0) 
+	    {
+                LOGWARN("Failed to Save DisplayMode to ssm_data\n");
+            }
+
+            tr181ErrorCode_t err = setLocalParam(rfc_caller_id, AVOUTPUT_ASPECTRATIO_RFC_PARAM, value.c_str());
+            if ( err != tr181Success ) 
+	    {
+                LOGWARN("setLocalParam for %s Failed : %s\n", AVOUTPUT_ASPECTRATIO_RFC_PARAM, getTR181ErrorString(err));
+            }
+            else 
+	    {
+                LOGINFO("setLocalParam for %s Successful, Value: %s\n", AVOUTPUT_ASPECTRATIO_RFC_PARAM, value.c_str());
+            }
+            LOGINFO("Exit : SetAspectRatio2() value : %s\n",value.c_str());
+            returnResponse(true);
+        }
+    }
+
+    uint32_t AVOutputTV::getZoomMode(const JsonObject& parameters, JsonObject& response)
+    {
+
+        LOGINFO("Entry\n");
+        tvDisplayMode_t mode;
+
```

Comment:
im not reviwing this implemetation in detials assuming it directly come from TV settings plugins

Reply:
API comes from tvsettings-plugin. But entire logic changed as per 
 https://etwiki.sys.comcast.net/display/RDKV/AVOutput+Plugin#AVOutputPlugin